### PR TITLE
feat: add fallback UI prop

### PR DESCRIPTION
**Pull request by IVLIU**
In actual development, We usually have to judge whether the array or object passed in is empty. If it is empty, we need to display the fallback UI.

### CODE REVIEW

**Comment by leebyron** [comment]
Hey @IVLIU, thanks for this suggestion. I'm curious, why did you close this PR after opening it?

**Comment by IVLIU** [comment]
Because of this failed build, so i reopen it.

File: react-loops.js [hunk 1]
```diff
@@ -79,19 +89,28 @@ function For(props) {
...
+  if(hasFallback) {
+    var fallback = props.fallback;
+  }
   if (hasIn) {
     // Get the object
     var obj = props.in;
 
     // Accept null / falsey as nothing to loop.
     if (!obj) {
+      console.warn(
```
Comment (leebyron) [inline react-loops.js]: In general I like to avoid direct use of `console.warn` in library code. It's easy to miss and when you are looking at your console it could lead to a lot of spew.

Also, the warning pattern is best saved for communicating deprecated usage patterns, and I didn't understand this PR's proposal as deprecating `null` behaving the same as a empty collection.

File: react-loops.js [hunk 2]
```diff
@@ -60,6 +60,16 @@ var iterall = require("iterall");
  *     }/>
  *   </ul>
  *
+ * * fallback in Loops*
+ * use the prop `fallback` to provide an fallback component, when we got a empty list.
+ * 
+ *   <ul>
+ *     <For of={myList}  fallback={_ => <div>fallback UI</div>}>
```
Comment (leebyron) [inline react-loops.js]: let's bike-shed on the name of this new prop. I think `fallback` makes sense, but are there alternatives we should consider? What do you think about `whenEmpty` or `ifEmpty`

File: react-loops.d.ts [hunk 3]
```diff
@@ -16,20 +16,24 @@ export function For<T>(
...
   props:
     | {
         in: O | null | undefined;
         as: ForCallback<O[K], K>;
+        fallback?: JSX.Element;
       }
     | {
         in: O | null | undefined;
         children: ForCallback<O[K], K>;
+        fallback?: JSX.Element;
```
Comment (leebyron) [inline react-loops.d.ts]: Thanks for adding types! Make sure to add these to the flow types as well

File: react-loops.d.ts [hunk 4]
```diff
@@ -16,20 +16,24 @@ export function For<T>(
     | {
         of: Iterable<T> | ArrayLike<T> | null | undefined;
         as: ForCallback<T, number>;
+        fallback?: JSX.Element;
```
Comment (leebyron) [inline react-loops.d.ts]: The type here says JSX.Element, but your examples and test case below show `() => JSX.Element`, which did you intend to propose?

File: react-loops.test.js [hunk 5]
```diff
@@ -89,7 +89,15 @@ describe("react-loops", () => {
         </>
       );
     });
-
+    it("loops fallback", () => {
```
Comment (leebyron) [inline react-loops.test.js]: Thanks for adding a test! Can you also test for `{}` in a `for-in` and `null` in both cases?

Also, what about a test if `fallback` is not a function?

**Comment by leebyron** [comment]
In general it's a good idea to run `yarn test` (or `npm test`) before sending a PR so you can have some confidence that the CI build will succeed. But also if the CI build fails but you still think your proposal has merit, you can comment that you're working on fixing the build or that you are hoping someone else will take the PR to completion (as an example, one of my favorite ways to submit a bug report issue on a project is to send a PR to add a test which will fail the build)

I just provided a code review, please let me know if you want to iterate on this or if you prefer that I should

File: react-loops.js (hunk 1 above)
Comment (IVLIU) [reply react-loops.js]: I can't agree with your point of view anymore. I will fix this

File: react-loops.js (hunk 2 above)
Comment (IVLIU) [reply react-loops.js]: I will reconsider this

File: react-loops.d.ts (hunk 4 above)
Comment (IVLIU) [reply react-loops.d.ts]: In fact, I hope this is a compatible version, but I accidentally missed it, I will fix it.

File: react-loops.test.js (hunk 5 above)
Comment (IVLIU) [reply react-loops.test.js]: I will fix it.

File: react-loops.js (hunk 1 above)
Comment (IVLIU) [reply react-loops.js]: In fact, I am a newbie and the first time I submitted the code. Thank you for your code review, I will modify it according to the suggestions.

**Comment by leebyron** [comment]
Any updates? I’m happy to help complete this if you don’t have the spare cycles

**Comment by IVLIU** [comment]
> Any updates? I’m happy to help complete this if you don’t have the spare cycles

Sorry, I am too busy recently. If you can achieve this, it will be great.

**Comment by leebyron** [comment]
Thanks for proposing this! It's now merged in and released
